renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -270,10 +278,125 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
     }
   }
 
+  // Returns true if the shader was modified.
+  bool StripSideEffects(const SPIRVPatchData &patchData, const char *entryName,
+                        std::vector<uint32_t> &modSpirv)
+  {
+    rdcspv::Editor editor(modSpirv);
+
+    editor.Prepare();
+
+    rdcspv::Id entryID;
+    for(const rdcspv::EntryPoint &entry : editor.GetEntries())
+    {
+      if(entry.name == entryName)
+      {
+        entryID = entry.id;
+        break;
+      }
+    }
+    bool modified = false;
+
+    std::set<rdcspv::Id> patchedFunctions;
+    std::set<rdcspv::Id> functionPatchQueue;
+    functionPatchQueue.insert(entryID);
+
+    while(!functionPatchQueue.empty())
+    {
+      rdcspv::Id funcId;
+      {
+        auto it = functionPatchQueue.begin();
+        funcId = *functionPatchQueue.begin();
+        functionPatchQueue.erase(it);
+        patchedFunctions.insert(funcId);
+      }
+
+      rdcspv::Iter it = editor.GetID(funcId);
+      RDCASSERT(it.opcode() == rdcspv::Op::Function);
+
+      it++;
+
+      for(; it; ++it)
+      {
+        rdcspv::Op opcode = it.opcode();
+        if(opcode == rdcspv::Op::FunctionEnd)
+          break;
+
+        switch(opcode)
+        {
+          case rdcspv::Op::FunctionCall:
+          {
+            rdcspv::OpFunctionCall call(it);
+            if(functionPatchQueue.find(call.function) == functionPatchQueue.end() &&
+               patchedFunctions.find(call.function) == patchedFunctions.end())
+              functionPatchQueue.insert(call.function);
+            break;
+          }
+          case rdcspv::Op::CopyMemory:
+          case rdcspv::Op::AtomicStore:
+          case rdcspv::Op::Store:
+          {
+            rdcspv::Id pointer = rdcspv::Id::fromWord(it.word(1));
+            rdcspv::Id pointerType = editor.GetIDType(pointer);
+            RDCASSERT(pointerType != rdcspv::Id());
+            auto pointerTypeIt = editor.GetID(pointerType);
```

Comment:
`rdcspv::Iter` here preferred to `auto` please since the type isn't horrible like stl iterators.